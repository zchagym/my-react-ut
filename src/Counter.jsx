import { useState } from 'react';
export default function Counter() {
    const [count, setCount] = useState(0);
    return (
        <div>
            <h1 data-testid="display">{count}</h1>
            <button type="button" onClick={() => setCount(count + 1)}>+1</button>
        </div>
    );
}
