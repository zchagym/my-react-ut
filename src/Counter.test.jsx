import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';

describe('Counter 组件', () => {
    it('初始显示为0', () => {
        render(<Counter />);
        expect(screen.getByTestId('display')).toHaveTextContent('0');
    });
    it('点击按钮计数+1', () => {
        render(<Counter />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByTestId('display')).toHaveTextContent('1');
    });
    it('Counter渲染快照', () => {
        const view = render(<Counter />);
        expect(view.asFragment()).toMatchSnapshot();
    });
    it('初始状态下，显示为0', () => {
        render(<Counter />);
        // 用 toBeInTheDocument 检查元素在页面中
        expect(screen.getByText('0')).toBeInTheDocument();
        // 用 getByRole 查询按钮，用户视角最佳实践
        expect(screen.getByRole('button', { name: '+1' })).toBeInTheDocument();
    });

    it('点击后计数递增，文本内容变化', () => {
        render(<Counter />);
        fireEvent.click(screen.getByRole('button', { name: '+1' }));
        expect(screen.getByTestId('display')).toHaveTextContent('1');
    });

    it('按钮包含正确属性', () => {
        render(<Counter />);
        const btn = screen.getByRole('button', { name: '+1' });
        expect(btn).toHaveAttribute('type', 'button'); // type为默认button
    });

    it('不可见元素不会被匹配', () => {
        render(<Counter />);
        // 这个文本不存在，所以会报错（用于体验断言失败情况），可注释掉体验
        // expect(screen.getByText('999')).toBeInTheDocument();
    });
});
