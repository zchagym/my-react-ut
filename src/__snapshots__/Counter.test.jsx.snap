// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter 组件 > Counter渲染快照 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="display"
    >
      0
    </h1>
    <button
      type="button"
    >
      +1
    </button>
  </div>
</DocumentFragment>
`;
